sdk/servicebus/Azure.Messaging.ServiceBus/MigrationGuide.md

```diff
@@ -0,0 +1,218 @@
+
+### Sending messages
+
+The v4 client allowed for sending a single message or a list of messages, which had the potential to fail unexpectedly if the maximum allowable size was exceeded. v7 aims to prevent this by asking that you first create a batch of messages using `CreateBatchAsync` and then attempt to add messages to that using `TryAdd()`. If the batch accepts a message, you can be confident that it will not violate size constraints when calling Send to send the batch.
+
+| In v4                                          | Equivalent in v7                                                 | Sample |
+|------------------------------------------------|------------------------------------------------------------------|--------|
+| `QueueClient.SendAsync(Message)` or `MessageSender.SendAsync(Message)`                          | `ServiceBusSender.SendAsync(ServiceBusMessage)`                               | [Send a message](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample01_HelloWorld.md#sending-and-receiving-a-message) |
+| `QueueClient.SendAsync(IList<Message>)` or `MessageSender.SendAsync(IList<Message>)`                          | `ServiceBusSender.SendBatchAsync(ServiceBusMessageBatch)`                               | [Send a batch of messages](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample01_HelloWorld.md#sending-and-receiving-a-batch-of-messages) |
+
+### Receiving messages 
+
+| In v4                                          | Equivalent in v7                                                 | Sample |
+|------------------------------------------------|------------------------------------------------------------------|--------|
+| `MessageReceiver.ReceiveAsync()`                      | `ServiceBusReceiver.ReceiveAsync()`                               | [Receive a message](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample01_HelloWorld.md#sending-and-receiving-a-message) |
+| `MessageReceiver.ReceiveAsync()`                      | `ServiceBusReceiver.ReceiveBatchAsync()`                               | [Receive a batch of messages](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample01_HelloWorld.md#sending-and-receiving-a-batch-of-messages) |
+| `QueueClient.RegisterMessageHandler()` or   `MessageReceiver.RegisterMessageHandler()`                    | `ServiceBusProcessor.StartProcessingAsync()`                               | [Receive messages using processor](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample04_Processor.md) |
```

Comment:
We should probably also include the code for setting the event handlers on the processor. We don't need to show what the event handler body is but at least show that it is necessary to set processor.ProcessMessageAsync to something.